pkg/agent/wireguard/interface.go

```diff
@@ -16,18 +16,24 @@ package wireguard
 
 import (
 	"net"
+
+	clientset "k8s.io/client-go/kubernetes"
 )
 
 type Interface interface {
 	// Init initializes the WireGuard client and sets up the WireGuard device.
-	// It will generate a new private key if necessary and update the public key to the Node's annotation.
-	Init() error
+	// It will generate a new private key if necessary.
+	// When "annotateNode" is true, it will update the public key to the Node's annotation.
+	// Or it will update the public key and port to Multi-cluster Gateway's wireGuard field.
```

Comment:
I don't see this is related to what the method actually does. Updating `Multi-cluster Gateway` is done on caller side.
I think we could make the process consistent for both cases: move the node annotating process to its caller side, so we don't need to pass annotateNode and clientSet as nil which looks weird for Multi-cluster case. The move makes sense as WireGuard Interface could focus on managing wireguard and let caller to decide how to publish the public key.

Reply:
Yes, I used to store `k8sClient` and `mcClient` in the WireGuard Client. The reason why we pass the `k8sClient` is that the client just used in `Init()`. But move the update step to the caller side will solve this, I will do it immediately. Thanks for advice.

Reply:
Move the update step to the caller side now. The `Init()` function just focuses on initializing WireGuard configuration and returning the public key.